### PR TITLE
Add support to pass extraContainers for sidecars

### PR DESCRIPTION
**Pull request by ekarlso**

### CODE REVIEW

File: charts/opentelemetry-collector/values.yaml [hunk 1]
```diff
@@ -284,6 +284,8 @@ lifecycleHooks: {}
 #       - /test/sleep
 #       - "5"
 
+extraContainers: []
+
```
Comment (dmitryax) [inline charts/opentelemetry-collector/values.yaml:288]: Can you please move it closer to `initContainers`?

**Review by dmitryax — COMMENTED** [review]
Please bump the version and run `make generate-examples`

**Comment by dmitryax** [comment]
Also values.schema.json needs to be updated

File: charts/opentelemetry-collector/values.yaml (hunk 1 above)
Comment (ekarlso) [reply charts/opentelemetry-collector/values.yaml:288]: Done.

Comment (ekarlso) [reply charts/opentelemetry-collector/values.yaml]: Done!

**Review by TylerHelmuth — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Pull request merged**